lib/client/api.go

```diff
@@ -527,7 +527,7 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error)
 		return trace.Wrap(err)
 	}
 	// Save profile to record proxy credentials
-	if err := tc.SaveProfile("", true); err != nil {
+	if err := tc.SaveProfile(tc.HomePath, true); err != nil {
```

Comment:
I noticed SaveProfile is always called with tc.HomePath (except here, where it's incorrect). Have you considered removing the param and moving the tc.Home logic inside it?

Reply:
@codingllama Yes, I was thinking about it. I think it would make sense to just use the `HomePath` internally instead of passing it every-time, but I think in this case we should also adjust `LoadProfile()` to make them consistent. The problem here is that `LoadProfile()` takes different input depending who runs it, `tsh` or `tctl`. 
https://github.com/gravitational/teleport/blob/56c536e61f4b52c011b7d18dfaaf2b2c9ecac1cc/tool/tctl/common/tctl.go#L412
Do you know what is the expected behaviour for `tctl`? Should it respect the `HomePath` or always default to `~/.tsh`?

Reply:
It's a good question. It seems to me that `tctl` should use `tc.HomePath` too, but we might want to check with someone else before we go and change that too - it is a slightly chage of behavior, after all.

Maybe  @russjones or @r0mant have an opinion?